### ISSUE TITLE
[libc++] Accepts invalid `struct _Atomic(T)` because `_Atomic` expands to a class template

### ISSUE THREAD

**Issue by eisenwave** [llvm/llvm-project#168579]
https://godbolt.org/z/97x9Ms8eK
```cpp
#include <stdatomic.h>
struct _Atomic(int) x;
```

This code should raise an error (or warning), but is falsely accepted by Clang.

https://eel.is/c++draft/stdatomic.h.syn specifies `_Atomic` as expanding to an exposition-only *`std-atomic`* alias template. This forbids the use of a `struct` keyword.

The bug occurs because libc++ defines `_Atomic` like:

```cpp
#define _Atomic(_Tp) ::std::atomic<_Tp>
```
https://github.com/llvm/llvm-project/blob/7c34848ae1405127a66abf10a963d8c7748f2e51/libcxx/include/stdatomic.h#L134-L138

See also https://gcc.gnu.org/bugzilla/show_bug.cgi?id=122738

**Comment by eisenwave** [comment]
This is now https://cplusplus.github.io/LWG/issue4599

The alias template is not actually an intended design aspect of the paper, so I would anticipate the LWG issue to be resolved as proposed.

CC @ldionne
